src/constructs/aws/Upload.ts

```diff
@@ -0,0 +1,179 @@
+                {
+                    allowedMethods: [HttpMethods.PUT],
+                    allowedOrigins: ["*"],
+                    allowedHeaders: ["*"],
+                },
+            ],
+            lifecycleRules: [
+                {
+                    expiration: Duration.days(1),
+                },
+            ],
+        });
+
+        this.bucketNameOutput = new CfnOutput(this, "BucketName", {
+            value: this.bucket.bucketName,
+        });
+
+        this.function = new LambdaFunction(this, "Function", {
+            code: Code.fromInline(this.createFunctionCode()),
+            handler: "index.handler",
+            runtime: Runtime.NODEJS_12_X,
+            environment: {
+                LIFT_UPLOAD_BUCKET_NAME: this.bucket.bucketName,
+            },
+            role: Role.fromRoleArn(
+                this,
+                "LambdaRole",
+                Fn.getAtt(this.provider.naming.getRoleLogicalId(), "Arn").toString()
+            ),
+        });
+
+        if (resolvedConfiguration.apiGateway === "http") {
+            this.provider.enableHttpApiCors();
+            this.httpApi = HttpApi.fromHttpApiAttributes(this, "HttpApi", {
+                httpApiId: Fn.ref(this.provider.naming.getHttpApiLogicalId()),
+            });
+
+            this.route = new HttpRoute(this, "Route", {
+                httpApi: this.httpApi,
+                integration: new LambdaProxyIntegration({
+                    handler: this.function,
+                }),
+                routeKey: HttpRouteKey.with("/upload-url", HttpMethod.POST),
+            });
+        }
+
+        if (resolvedConfiguration.apiGateway === "rest") {
+            this.restApi = RestApi.fromRestApiAttributes(this, "RestApi", {
+                restApiId: Fn.ref(this.provider.naming.getRestApiLogicalId()),
+                rootResourceId: Fn.getAtt(this.provider.naming.getRestApiLogicalId(), "RootResourceId").toString(),
+            }) as RestApi;
+
+            const resource: Resource = this.restApi.root.addResource("upload-url");
+            resource.addCorsPreflight({
+                allowHeaders: ["*"],
+                allowMethods: ["POST"],
+                allowOrigins: ["*"],
+            });
+            resource.addMethod("POST", new LambdaIntegration(this.function));
+        }
+    }
+
+    variables(): Record<string, unknown> {
+        return {
+            bucketArn: this.bucket.bucketArn,
+            bucketName: this.bucket.bucketName,
+        };
+    }
+
+    permissions(): PolicyStatement[] {
+        return [
+            new PolicyStatement(
+                ["s3:PutObject", "s3:GetObject", "s3:DeleteObject", "s3:ListBucket"],
+                [this.bucket.bucketArn, Stack.of(this).resolve(Fn.join("/", [this.bucket.bucketArn, "*"]))]
+            ),
+        ];
+    }
+
+    outputs(): Record<string, () => Promise<string | undefined>> {
+        return {
+            bucketName: () => this.getBucketName(),
```

Comment:
it would be nice to output the full upload URL but I wasn't able to generate it properly for Rest APIs. Another problem was that people may have domains configured on their API Gateway or they may be using the server-side construct which woul make the outputed URL useless.